tests/service/ExprTests.fs

```diff
@@ -918,43 +959,72 @@ let ``Test Optimized Declarations Project1`` () =
 
     printDeclarations None (List.ofSeq file1.Declarations) 
       |> Seq.toList 
-      |> filterHack
-      |> shouldEqual (filterHack expected)
+      |> Utils.filterHack
+      |> shouldPairwiseEqual (Utils.filterHack expected)
 
     printDeclarations None (List.ofSeq file2.Declarations) 
       |> Seq.toList 
-      |> filterHack
-      |> shouldEqual (filterHack expected2)
+      |> Utils.filterHack
+      |> shouldPairwiseEqual (Utils.filterHack expected2)
 
     ()
 
+[<Test>]
+let ``Test Optimized and Unoptimized Declarations for Project1`` () =
+    let filenames = fst Project1.options.Value
+    try
+        ``Test Optimized Declarations Project1`` ()
+        ``Test Unoptimized Declarations Project1`` ()
+    finally
+        Utils.cleanupTempFiles filenames
+
 let testOperators dnName fsName excludedTests expectedUnoptimized expectedOptimized =
-    let basePath = Path.GetTempFileName()
-    let fileName = Path.ChangeExtension(basePath, ".fs")
-    let dllName = Path.ChangeExtension(basePath, ".dll")
-    let projFileName = Path.ChangeExtension(basePath, ".fsproj")
-    let source = System.String.Format(Project1.operatorTests, dnName, fsName)
-    let replace (s:string) r = s.Replace("let " + r, "// let " + r)
-    let fileSource = excludedTests |> List.fold replace source
-    File.WriteAllText(fileName, fileSource)
-
-    let args = mkProjectCommandLineArgsSilent (dllName, [fileName])
-    let options =  checker.GetProjectOptionsFromCommandLineArgs (projFileName, args)
-    let wholeProjectResults = exprChecker.ParseAndCheckProject(options) |> Async.RunSynchronously
 
-    for e in wholeProjectResults.Errors do 
-        printfn "%s Operator Tests error: <<<%s>>>" dnName e.Message
+    let tempFileName = Utils.getTempFileName()
+    let filePath = Utils.getTempFilePathChangeExt tempFileName ".fs"
+    let dllPath =Utils.getTempFilePathChangeExt tempFileName ".dll"
+    let projFilePath = Utils.getTempFilePathChangeExt tempFileName ".fsproj"
 
-    wholeProjectResults.Errors.Length |> shouldEqual 0
+    try
+        createTempDir()
+        let source = System.String.Format(Project1.operatorTests, dnName, fsName)
+        let replace (s:string) r = s.Replace("let " + r, "// let " + r)
+        let fileSource = excludedTests |> List.fold replace source
+        File.WriteAllText(filePath, fileSource)
+
+        let args = [|
+            yield! mkProjectCommandLineArgsSilent (dllPath, [filePath])
+            yield @"-r:System.Numerics.dll"         // needed for some tests
```

Comment:
This is a desktop framework .dll, does this test only build desktop ?

Reply:
No, though it took me a week to find out, but it is run both by framework and dotnet. And with old FSharp.Core.dll, but that's something I found out in the string PR.

It doesn't complain about it, though. @kevinransom Should I only add it when framework is run?

Reply:
Here would probably be best:  https://github.com/dotnet/fsharp/blob/master/tests/service/Common.fs#L90

````
let mkStandardProjectReferences () = 
#if NETCOREAPP
            let file = "Sample_NETCoreSDK_FSharp_Library_netstandard2_0.fsproj"
            let projDir = Path.Combine(__SOURCE_DIRECTORY__, "../projects/Sample_NETCoreSDK_FSharp_Library_netstandard2_0")
            readRefs projDir file
#else
            [ yield sysLib "mscorlib"
              yield sysLib "System"
              yield sysLib "System.Core"
              yield sysLib "System.Numerics"
              yield fsCoreDefaultReference() ]
#endif   
````

Reply:
Done, makes sense indeed (I didn't like my hacky attempt anyway ;). Let's see if it passes the tests still.